src/python.cc

```diff
@@ -361,8 +421,394 @@ class ModelInstanceState : public BackendModelInstance {
+          (std::string("Path ") + *path_to_activate_ +
+           " does not exist. The Python environment should contain an "
+           "'activate' script.")
+              .c_str());
+    }
+  }
+
+  (*parent_pid_) = getpid();
+
+  // Destruct any in-use shared memory object before starting the stub process.
+  (*ipc_control_) = nullptr;
+  (*stub_message_queue_) = nullptr;
+  (*parent_message_queue_) = nullptr;
+  (*memory_manager_) = nullptr;
+  if (isInstance) {
+    (*thread_pool_) = std::make_unique<boost::asio::thread_pool>(
+        model_state->StateForBackend()->thread_pool_size);
+  }
+  int64_t shm_default_size =
+      model_state->StateForBackend()->shm_default_byte_size;
+  int64_t shm_growth_byte_size =
+      model_state->StateForBackend()->shm_growth_byte_size;
+
+  try {
+    // It is necessary for restart to make sure that the previous shared memory
+    // pool is destructed before the new pool is created.
+    (*shm_pool_) = nullptr;
+    (*shm_pool_) = std::make_unique<SharedMemoryManager>(
+        (*shm_region_name_), shm_default_size, shm_growth_byte_size,
+        true /* create */);
+  }
+  catch (const PythonBackendException& pb_exception) {
+    return TRITONSERVER_ErrorNew(
+        TRITONSERVER_ERROR_INTERNAL, pb_exception.what());
+  }
+
+  AllocatedSharedMemory<IPCControlShm> ipc_control =
+      (*shm_pool_)->Construct<IPCControlShm>();
+  (*ipc_control_) = std::move(ipc_control.data_);
+  (*ipc_control_handle_) = ipc_control.handle_;
+
+  auto message_queue_size =
+      model_state->StateForBackend()->shm_message_queue_size;
+
+  RETURN_IF_EXCEPTION(
+      (*stub_message_queue_) =
+          MessageQueue<bi::managed_external_buffer::handle_t>::Create(
+              (*shm_pool_), message_queue_size));
+  RETURN_IF_EXCEPTION(
+      (*parent_message_queue_) =
+          MessageQueue<bi::managed_external_buffer::handle_t>::Create(
+              (*shm_pool_), message_queue_size));
+
+  std::unique_ptr<MessageQueue<intptr_t>> memory_manager_message_queue;
+  RETURN_IF_EXCEPTION(
+      memory_manager_message_queue =
+          MessageQueue<intptr_t>::Create((*shm_pool_), message_queue_size));
+
+  memory_manager_message_queue->ResetSemaphores();
+  (*ipc_control_)->memory_manager_message_queue =
+      memory_manager_message_queue->ShmHandle();
+  (*ipc_control_)->decoupled = model_state->IsDecoupled();
+
+  (*memory_manager_) =
+      std::make_unique<MemoryManager>(std::move(memory_manager_message_queue));
+  (*ipc_control_)->parent_message_queue = (*parent_message_queue_)->ShmHandle();
+  (*ipc_control_)->stub_message_queue = (*stub_message_queue_)->ShmHandle();
+
+  new (&((*ipc_control_)->stub_health_mutex)) bi::interprocess_mutex;
+  (*health_mutex_) = &((*ipc_control_)->stub_health_mutex);
+
+  (*stub_message_queue_)->ResetSemaphores();
+  (*parent_message_queue_)->ResetSemaphores();
+
+  (*initialized_) = false;
+
+  return nullptr;
+}
+
+TRITONSERVER_Error*
+StubLauncher::StubProcess(
```

Comment:
StubProcess -> Launch

Also, can you make forking the process part of launching the stub process too?